### PR TITLE
Develop react_list-of-goods

### PR DESCRIPTION
**Pull request by sergey-lipikhin**
[DEMO LINK](https://sergey-lipikhin.github.io/react_list-of-goods/)

### CODE REVIEW

**Review by Anton-Tretiak — APPROVED** [review]
Well done!

File: src/App.tsx [hunk 1]
```diff
@@ -26,70 +27,132 @@ type ReorderOptions = {
...
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort();
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+      break;
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:42]: Let's move switch/case inside sort method, why duplicating it?

File: src/App.tsx [hunk 2]
```diff
@@ -26,70 +27,132 @@ type ReorderOptions = {
...
+          {isResetButtonVisible && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => (
+                this.setState({
+                  sortType: SortType.NONE,
+                  isReversed: false,
+                })
+              )}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:136]: Let's create `resetSorting` method, that will set initial values to state

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -26,70 +27,136 @@ type ReorderOptions = {
...
+            Reverse
+          </button>
+
+          {isResetButtonVisible && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => (
+                this.resetSorting()
+              )}
```
Comment (sergii-nosachenko) [inline src/App.tsx:140]:
```suggestion
              onClick={this.resetSorting}
```

File: src/App.tsx [hunk 4]
```diff
@@ -26,70 +27,136 @@ type ReorderOptions = {
...
+            {reorderedGoods.map(good => (
+              <li
+                data-cy="Good"
+                key={good}
+              >
+                {good}
+              </li>
+            ))}
+          </ul>
         </ul>
```
Comment (sergii-nosachenko) [inline src/App.tsx:158]: Can be a separate component

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 5]
```diff
@@ -62,12 +63,12 @@ export class App extends React.Component<{}, State> {
     isReversed: false,
   };
 
-  resetSorting() {
+  resetSorting = () => {
```
Comment (sergii-nosachenko) [inline src/App.tsx:66]:
```suggestion
  resetSorting() {
```

File: src/components/GoodsList/GoodsList.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+type Props = {
+  goods: string[],
+};
+
+export const GoodsList: React.FC<Props> = ({ goods }) => {
+  return (
```
Comment (sergii-nosachenko) [inline src/components/GoodsList/GoodsList.tsx:8]: Useless `return`

**Review by sergii-nosachenko — APPROVED** [review]
Good work 👍
Commented

**Pull request closed without merging**